File: template/go/part1.go
package main

import (
	"fmt"
	"log"
)

func solve1(input string) int {
	result := 0

	lines, err := readLines(input)
	if err != nil {
		log.Fatal(err)
	}

	for _, line := range lines {
		fmt.Printf("Line from file: %s\n", line)
	}

	return result
}

File: template/c/part1.c
#include <stdio.h>

int solve1(const char *filename) {
    FILE *file = fopen(filename, "r");
    if (file == NULL) {
        perror("Error opening file");
        return -1;
    }

    int result = 0;
    char line[100];

    while (fgets(line, sizeof(line), file) != NULL) {
        printf("%s\n", line);
    }

    fclose(file);
    return result;
}